plugins/parquet/parquetembed.cpp

```diff
@@ -770,10 +770,10 @@ arrow::Status ParquetWriter::fieldToNode(const RtlFieldInfo *field, std::vector<
     case type_unicode:
     case type_varunicode:
     case type_decimal:
-        arrowFields.push_back(std::make_shared<arrow::Field>(name.str(), arrow::utf8())); //TODO add decimal encoding
+        arrowFields.push_back(std::make_shared<arrow::Field>(name.str(), arrow::utf8())); // TODO: add decimal encoding
         break;
     case type_data:
-        arrowFields.push_back(std::make_shared<arrow::Field>(name.str(), arrow::large_binary()));
+        arrowFields.push_back(std::make_shared<arrow::Field>(name.str(), arrow::fixed_size_binary(field->type->length)));
```

Comment:
I don't think we should be writing fixed_size_binary to Parquet files in all cases. I don't know the answer to this question, but what is the value of field->type->length if the DATA field is defined without the number of bytes e.g. DATA mydata vs DATA8 myfixeddata.

Reply:
Good question. I will look into this and get back to you.